src/stability_sdk/animation.py

```diff
@@ -420,16 +422,46 @@ def generate_depth_image(self, image: Image.Image) -> Image.Image:
         return results[0]
 
     def get_animation_prompts_weights(self, frame_idx: int) -> Tuple[List[str], List[float]]:
+        prev, next, tween = self.get_key_frame_tween(frame_idx)
+        if prev == next or not self.args.interpolate_prompts:
+            return [self.animation_prompts[prev]], [1.0]
+        else:
+            return [self.animation_prompts[prev], self.animation_prompts[next]], [1.0 - tween, tween]
+
+    def get_color_match_image(self, frame_idx: int) -> Image.Image:
+        if not self.args.color_match_animate:
+            return self.color_match_images.get(0)
+
+        prev, next, tween = self.get_key_frame_tween(frame_idx)
+
+        if prev not in self.color_match_images:
+            self.color_match_images[prev] = self._span_render_frame(prev, self.args.seed)
```

Comment:
You use `_span_render_frame` just because it incapsulates all the cumbersome generation parameter setup that you need, right? Should it be called differently then? Smth like just `_generate_frame`? I worry it might confuse since it's used not only in the span_cadence mode.

Reply:
Yeah I think that's a good idea`_span_render_frame` -> `_render_frame` standalone simple frame render function used internally.